api/v1beta1/azuremanagedcontrolplane_webhook_test.go

```diff
@@ -2683,3 +2770,670 @@ func getAMCPMetaData() metav1.ObjectMeta {
+							Enabled:       true,
+							IntervalHours: ptr.To(28),
+						},
+					},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "AzureManagedControlPlane SecurityProfile.ImageCleaner cannot be unset",
+			oldAMCP: &AzureManagedControlPlane{
+				Spec: AzureManagedControlPlaneSpec{
+					AzureManagedControlPlaneClassSpec: AzureManagedControlPlaneClassSpec{
+						Version: "v1.18.0",
+					},
+					SecurityProfile: &ManagedClusterSecurityProfile{
+						ImageCleaner: &ManagedClusterSecurityProfileImageCleaner{
+							Enabled:       true,
+							IntervalHours: ptr.To(48),
+						},
+					},
+				},
+			},
+			amcp: &AzureManagedControlPlane{
+				Spec: AzureManagedControlPlaneSpec{
+					AzureManagedControlPlaneClassSpec: AzureManagedControlPlaneClassSpec{
+						Version: "v1.18.0",
+					},
+					SecurityProfile: &ManagedClusterSecurityProfile{},
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "AzureManagedControlPlane SecurityProfile.ImageCleaner is mutable",
+			oldAMCP: &AzureManagedControlPlane{
+				Spec: AzureManagedControlPlaneSpec{
+					AzureManagedControlPlaneClassSpec: AzureManagedControlPlaneClassSpec{
+						Version: "v1.18.0",
+					},
+					SecurityProfile: &ManagedClusterSecurityProfile{
+						ImageCleaner: &ManagedClusterSecurityProfileImageCleaner{
+							Enabled: true,
+						},
+					},
+				},
+			},
+			amcp: &AzureManagedControlPlane{
+				Spec: AzureManagedControlPlaneSpec{
+					AzureManagedControlPlaneClassSpec: AzureManagedControlPlaneClassSpec{
+						Version: "v1.18.0",
+					},
+					SecurityProfile: &ManagedClusterSecurityProfile{
+						ImageCleaner: &ManagedClusterSecurityProfileImageCleaner{
+							IntervalHours: ptr.To(48),
+						},
+					},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "AzureManagedControlPlane SecurityProfile.ImageCleaner can be disabled",
+			oldAMCP: &AzureManagedControlPlane{
+				Spec: AzureManagedControlPlaneSpec{
+					AzureManagedControlPlaneClassSpec: AzureManagedControlPlaneClassSpec{
+						Version: "v1.18.0",
+					},
+					SecurityProfile: &ManagedClusterSecurityProfile{
+						ImageCleaner: &ManagedClusterSecurityProfileImageCleaner{
+							Enabled:       true,
+							IntervalHours: ptr.To(48),
+						},
+					},
+				},
+			},
+			amcp: &AzureManagedControlPlane{
+				Spec: AzureManagedControlPlaneSpec{
+					AzureManagedControlPlaneClassSpec: AzureManagedControlPlaneClassSpec{
+						Version: "v1.18.0",
+					},
+					SecurityProfile: &ManagedClusterSecurityProfile{
+						ImageCleaner: &ManagedClusterSecurityProfileImageCleaner{
+							Enabled:       false,
+							IntervalHours: ptr.To(36),
+						},
+					},
+				},
+			},
+			wantErr: false,
+		},
+	}
+	client := mockClient{ReturnError: false}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			g := NewWithT(t)
+			mcpw := &azureManagedControlPlaneWebhook{
+				Client: client,
+			}
+			_, err := mcpw.ValidateUpdate(context.Background(), tc.oldAMCP, tc.amcp)
```

Comment:
ditto as above re: do these test cases not fit into an existing table that calls `ValidateUpdate`?

Reply:
Same as above did not, the test functions are too crowded and big wanted to split it to keep it clean.